pkg/cmd/gtctl/connect/mysql/mysql.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysql
+
+import (
+	"errors"
+	"github.com/spf13/cobra"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+const (
+	UpSeparator    = ":"
+	AtSeparator    = "@"
+	ArgHost        = "-h"
+	ArgPort        = "-P"
+	ArgUser        = "-u"
+	ArgPassword    = "-p"
+	MySQL          = "mysql"
+	SplitSeparator = "://"
+)
+
+func ConnectCommand(cmd *cobra.Command, args []string) error {
+	s := args[0]
+	suffix := strings.Split(s, SplitSeparator)[1]
+	split := strings.Split(suffix, AtSeparator)
+	if len(split) != 2 {
+		return errors.New("invalid argument, you can try gtctl connect mysql://user:password@host:port")
+	}
+	up := strings.Split(split[0], UpSeparator)
+	if len(up) != 2 {
+		return errors.New("invalid argument, you can try gtctl connect mysql://user:password@host:port")
+	}
+	hp := strings.Split(split[1], UpSeparator)
+	if len(hp) != 2 {
+		return errors.New("invalid argument, you can try gtctl connect mysql://user:password@host:port")
+	}
+	user := up[0]
+	password := up[1]
+	host := hp[0]
+	port := hp[1]
+	return Connect(user, password, host, port)
+}
+
+func Connect(user, password, host, port string) error {
```

Comment:
question: Should the user need to execute `kubectl port-forward` if the cluster is deployed in K8s?

Reply:
I will think about it!

Reply:
@sjcsjc123 I can give you my try(it's not so good, but it works):

```go
package main

import (
	"context"
	"database/sql"
	"github.com/go-sql-driver/mysql"
	"os"
	"os/exec"
	"sync"
)

type Connection struct {
	wg sync.WaitGroup

	cancelPortforward context.CancelFunc
}

func main() {
	c := Connection{}
	ctx, cancel := context.WithCancel(context.Background())
	c.cancelPortforward = cancel

	portForwardCmd := exec.CommandContext(ctx, "kubectl", "port-forward", "-n", "default", "svc/mycluster-frontend", "4002:4002")
	portForwardCmd.Stderr = os.Stderr
	portForwardCmd.Stdout = os.Stdout

	err := portForwardCmd.Start()
	if err != nil {
		panic(err)
	}

	go func() {
		c.wg.Add(1)
		defer c.wg.Done()
		if err = portForwardCmd.Wait(); err != nil {
			// Maybe we should not panic here?
			// Will output some error message to the console when the port-forward is canceled.
			panic(err)
		}
	}()

	// Wait for port-forward to be ready.
	for {
		cfg := mysql.Config{
			Net:                  "tcp",
			Addr:                 "127.0.0.1:4002",
			User:                 "",
			Passwd:               "",
			DBName:               "",
			AllowNativePasswords: true,
		}

		db, err := sql.Open("mysql", cfg.FormatDSN())
		if err != nil {
			continue
		}

		_, err = db.Conn(context.TODO())
		if err != nil {
			continue
		}

		db.Close()
		break
	}

	connectMySQLCmd := exec.Command("mysql", "-h", "127.0.0.1", "-P", "4002")
	connectMySQLCmd.Stderr = os.Stderr
	connectMySQLCmd.Stdout = os.Stdout
	connectMySQLCmd.Stdin = os.Stdin
	if err = connectMySQLCmd.Start(); err != nil {
		panic(err)
	}

	if err = connectMySQLCmd.Wait(); err != nil {
		panic(err)
	}

	c.cancelPortforward()
	c.wg.Wait()
}

```

Reply:
Thanks!